tests/integ/scala/test_dataframe_writer_suite.py

```diff
@@ -0,0 +1,203 @@
+    Utils.create_table(
+        session, table_name, "a int, b int, c int autoincrement", is_temporary=True
+    )
+    try:
+        df1 = session.create_dataframe([[1, 2]], schema=["b", "a"])
+        df1.write.save_as_table(
+            table_name, mode="append", column_order="name", table_type="temp"
+        )
+        Utils.check_answer(session.table(table_name), [Row(2, 1, 1)])
+    finally:
+        Utils.drop_table(session, table_name)
+
+
+def test_negative_write_with_target_column_name_order(session):
+    table_name = Utils.random_table_name()
+    Utils.create_table(session, table_name, "a int, b int", is_temporary=True)
+    try:
+        df1 = session.create_dataframe([[1, 2]], schema=["a", "c"])
+        # The "columnOrder = name" needs the DataFrame has the same column name set
+        with pytest.raises(SnowparkSQLException, match="invalid identifier 'C'"):
+            df1.write.save_as_table(
+                table_name, mode="append", column_order="name", table_type="temp"
+            )
+
+        with pytest.raises(
+            ValueError, match="'column_order' must be either 'name' or 'index'"
+        ):
+            df1.write.save_as_table(
+                table_name, mode="append", column_order="any_value", table_type="temp"
+            )
+    finally:
+        Utils.drop_table(session, table_name)
+
+
+def test_write_with_target_column_name_order_all_kinds_of_dataframes(
+    session, resources_path
+):
+    table_name = Utils.random_table_name()
+    Utils.create_table(session, table_name, "a int, b int", is_temporary=True)
+    try:
+        df1 = session.create_dataframe([[1, 2]], schema=["b", "a"])
+        # DataFrame.cache_result()
+        df_cached = df1.cache_result()
+        df_cached.write.save_as_table(
+            table_name, mode="append", column_order="name", table_type="temp"
+        )
+        Utils.check_answer(session.table(table_name), [Row(2, 1)])
+
+        # copy DataFrame
+        session._conn.run_query(f"truncate table {table_name}")
+        df_cloned = copy.copy(df1)
+        df_cloned.write.save_as_table(
+            table_name, mode="append", column_order="name", table_type="temp"
+        )
+        Utils.check_answer(session.table(table_name), [Row(2, 1)])
+
+        # large local relation
+        session._conn.run_query(f"truncate table {table_name}")
+        large_df = session.create_dataframe([[1, 2]] * 1024, schema=["b", "a"])
+        large_df.write.save_as_table(
+            table_name, mode="append", column_order="name", table_type="temp"
+        )
+        rows = session.table(table_name).collect()
+        assert len(rows) == 1024
+        for row in rows:
+            assert row["B"] == 1 and row["A"] == 2
+    finally:
+        Utils.drop_table(session, table_name)
+
+    # show tables
+    # Create a DataFrame from SQL `show tables` and then filter on it not supported yet. Enable the following test after it's supported.
```

Comment:
jira?

Reply:
We shouldn't paste a JIRA link here. It's a URL not accessible to the public.

Reply:
Then maybe a github issue?

Reply:
I created a JIRA. Let's discuss later whether we use github to manage these issues.